x/globalfee/ante/fee_utils.go

```diff
@@ -96,17 +111,72 @@ func IsAllGT(a, b sdk.DecCoins) bool {
 	return true
 }
 
-func DenomsSubsetOf(a, b sdk.DecCoins) bool {
-	// more denoms in B than in a
+// DenomsSubsetOf checks if the denominations in a are a subset of those in b.
+func DenomsSubsetOfG[T any, A number[A], S coinSlice[T, A]](a, b S) bool {
 	if len(a) > len(b) {
 		return false
 	}
 
-	for _, coin := range a {
-		if b.AmountOf(coin.Denom).IsZero() {
+	for idx := range a {
+		if b.AmountOf(a.GetDenomByIndex(idx)).IsZero() {
 			return false
 		}
 	}
 
 	return true
 }
+
+// checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
+// unit of gas is fixed and set by each validator, can the tx priority is computed from the gas price.
+func CheckTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return nil, 0, errorsmod.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	feeCoins := feeTx.GetFee()
+	gas := feeTx.GetGas()
+
+	// Ensure that the provided fees meet a minimum threshold for the validator,
+	// if this is a CheckTx. This is only for local mempool purposes, and thus
+	// is only ran on check tx.
+	minGasPrices := ctx.MinGasPrices()
```

Comment:
@edjroz why are we running this logic without the checkTx flag ? does it need to run in deliverTx ?

Reply:
This way we ensure that the transaction always verify `ctx.MinGasPrices` and calculates accordingly. it is still on draft stage, so suggestions are welcomed here.

Reply:
@edjroz what I am saying is, this check is ensuring that the tx set an acceptable fee right ? this check is only required during checkTx stage since that runs before the deliverTx stage. At the deliverTx stage, the check that the tx has the proper fees set must have already happened so there is no need to re-run the logic again in deliverTx